### ISSUE TITLE
Improve Modal accessibility

### ISSUE THREAD

**Issue by trilodge** [SAP/fundamental-vue#236]
#### Enhancement

Modal dialog should be improved according to this article + demo 
https://bitsofco.de/accessible-modal-dialog/ 

Current version lacks a aria-role="dialog" attribute and aria-labelledby="" and focus handling should be added. Closing the modal with ESC should also be added to further improve on keyboard navigation. 

Focus handling means, the element which called the modal should be remembered and after closing the modal the keyboard focus must return to that element. While opened modal keyboard focus needs to be shifted with tabindex to the first element of the modal itself. 

#### Which versions of Vue.JS and Fundamental Vue are affected? (If this is a feature request, use current version.)

Current version

**Comment by ChristianKienle** [comment]
Just want to mention that in one of the last calls this was mentioned:

https://github.com/davidtheclark/focus-trap

We could think about integrating this library which will not solve all problems but improve keyboard navigation.

**Comment by trilodge** [comment]
Interesting. From what i read there in the description that would be the perfect fit.

**Comment by albrechtjohannes** [comment]
I added the mentioned library and added it to the modal
